fv3gfs/physics/physics_state.py

```diff
@@ -1,75 +1,134 @@
-            "cloud_water_mixing_ratio",
-            "rain_mixing_ratio",
-            "cloud_ice_mixing_ratio",
-            "snow_mixing_ratio",
-            "graupel_mixing_ratio",
-            "ozone_mixing_ratio",
-            "turbulent_kinetic_energy",
-            "cloud_fraction",
-            "eastward_wind",
-            "northward_wind",
-            "vertical_wind",
-            "pressure_thickness_of_atmospheric_layer",
-            "vertical_thickness_of_atmospheric_layer",
-            "vertical_pressure_velocity",
-        ]
-        phy_state = {key: copy.deepcopy(dycore_state[key]) for key in phy_vars}
-        return phy_state
+    @classmethod
+    def from_dycore_state(cls, state, storage: FloatField) -> "PhysicsState":
+        """
+        Constructor for PhysicsState when using dynamical core state
+        storage: storage for variables not in dycore
+        """
+        # [TODO] we may want to use a copy from dycore instead of using the same storages
+        return cls(
+            qvapor=state.qvapor,
+            qliquid=state.qliquid,
+            qrain=state.qrain,
+            qsnow=state.qsnow,
+            qice=state.qice,
+            qgraupel=state.qgraupel,
+            qo3mr=state.qo3mr,
+            qsgs_tke=state.qsgs_tke,
+            qcld=state.qcld,
+            pt=state.pt,
+            delp=state.delp,
+            delz=state.delz,
+            ua=state.ua,
+            va=state.va,
+            w=state.w,
+            omga=state.omga,
+            delprsi=copy.deepcopy(storage),
+            phii=copy.deepcopy(storage),
+            phil=copy.deepcopy(storage),
+            dz=copy.deepcopy(storage),
+            wmp=copy.deepcopy(storage),
+            qvapor_t1=copy.deepcopy(storage),
+            qliquid_t1=copy.deepcopy(storage),
+            qrain_t1=copy.deepcopy(storage),
+            qsnow_t1=copy.deepcopy(storage),
+            qice_t1=copy.deepcopy(storage),
+            qgraupel_t1=copy.deepcopy(storage),
+            qcld_t1=copy.deepcopy(storage),
+            pt_t1=copy.deepcopy(storage),
+            ua_t1=copy.deepcopy(storage),
+            va_t1=copy.deepcopy(storage),
+        )
 
-    @property
-    def microphysics(self) -> MicrophysicsState:
-        microphysics_state = MicrophysicsState(
-            self.grid,
-            self.physics_state.pt_quantity,
-            self.physics_state.qvapor_quantity,
-            self.physics_state.qliquid_quantity,
-            self.physics_state.qrain_quantity,
-            self.physics_state.qice_quantity,
-            self.physics_state.qsnow_quantity,
-            self.physics_state.qgraupel_quantity,
-            self.physics_state.qcld_quantity,
-            self.physics_state.ua_quantity,
-            self.physics_state.va_quantity,
-            self.physics_state.delp_quantity,
-            self.physics_state.delz_quantity,
-            self.physics_state.omga_quantity,
+    def microphysics(self, storage) -> MicrophysicsState:
```

Comment:
Why does this need to take in a `storage` argument? Shouldn't `PhysicsState` already have everything needed to make a `MicrophysicsState`?

Reply:
`storage` is used for variables exclusively inside `MicrophysicsState`. In this case, tendencies as a result of microphysics are initialized using `storage` (e.g., `qv_dt`, `ql_dt`). `PhysicsState` also has tendencies, but they are tendencies from physics (sum of all physics schemes). Does it make sense to keep this distinction? or should we label `qv_dt_microph` in `PhysicsState`?